### PR TITLE
{AKS} az aks update: enable outbound migration from/to udr in BYO vnet scenario

### PR DESCRIPTION
**Pull request by MartinForReal**
---

This checklist is used to make sure that common guidelines for a pull request are followed.

### General Guidelines

- [x] Have you run `azdev style <YOUR_EXT>` locally? (`pip install azdev` required)
- [x] Have you run `python scripts/ci/test_index.py -q` locally?

For new extensions:

- [x] My extension description/summary conforms to the [Extension Summary Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/extensions/extension_summary_guidelines.md).

### About Extension Publish

There is a pipeline to automatically build, upload and publish extension wheels.  
Once your pull request is merged into main branch, a new pull request will be created to update `src/index.json` automatically.  
The precondition is to put your code inside this repository and upgrade the version in the pull request but do not modify `src/index.json`.

### CODE REVIEW

**Review by FumingZhang — COMMENTED** [review]
Could you please help add some unit test cases?

File: src/aks-preview/azext_aks_preview/managed_cluster_decorator.py [hunk 1]
```diff
@@ -321,7 +320,7 @@ def _get_outbound_type(
                             "be pre-configured with a route table with egress rules"
                         )
 
-                if outbound_type == CONST_OUTBOUND_TYPE_USER_DEFINED_ROUTING:
+                if self.decorator_mode != DecoratorMode.UPDATE and outbound_type == CONST_OUTBOUND_TYPE_USER_DEFINED_ROUTING:
```
[the remark this answers is not in the document]
Comment (MartinForReal) [reply src/aks-preview/azext_aks_preview/managed_cluster_decorator.py]: Thanks! updated the code.

File: src/aks-preview/azext_aks_preview/managed_cluster_decorator.py [hunk 2]
```diff
@@ -312,7 +312,6 @@ def _get_outbound_type(
                     )
 
                 if outbound_type in [
-                    CONST_OUTBOUND_TYPE_USER_DEFINED_ROUTING,
                     CONST_OUTBOUND_TYPE_USER_ASSIGNED_NAT_GATEWAY,
                 ]:
                     if self.get_vnet_subnet_id() in ["", None]:
```
[the remark this answers is not in the document]
Comment (MartinForReal) [reply src/aks-preview/azext_aks_preview/managed_cluster_decorator.py]: this change has been reverted

**Review by FumingZhang — APPROVED** [review]
LGTM

**Comment by yonzhan** [comment]
AKS

**Review by zhoxing-ms — APPROVED** [review]

**Pull request merged**